### ISSUE TITLE
why are we not showing physical web icon in notification panel if physical web is nearby?

### ISSUE THREAD

**Issue by kanodia** [google/physical-web#650]
Hello

We show physical web icon only once at the run time permission. But when for the second time if a user comes under the range of a URL, there is no indication on te screen for a user to know that he has physical web.

I understand we cannot use sound or vibration as an indicator but why are we not even showing the physical web icon in the status bar?

**Comment by scottjenson** [comment]
I'm trying to understand you question. Are you saying that it *IS* showing up in as a notification, but it is *NOT* showing up in the status bar?

**Comment by kanodia** [comment]
No 

I am saying that physical web icon does not appear in status bar. When we pull down the notification panel, we can see physical web pages nearby but there is no indication of it on status bar.


For example, if there is a whatsapp notification, then we see a whatsapp icon on the status bar. It helps us in identifying that if we pull down the notification panel, we will see a whatsapp notification.

This kind of behavior is absent in physical web.

**Comment by scottjenson** [comment]
That is a function of the priority of the notification. At this point in time, the Physical Web notification is low priority so doesn't show up in the status bar. We understand this is a concern and we're experimenting with alternative approaches.

**Comment by kanodia** [comment]
I have a suggestion. Say if a user has enabled physical web and have accessed it once, then may be it can turn into a priority notification for the particular user.

Another suggestion: If a user has accessed a particular domain(URL) through physical web once(or may be we can define a threshold) then if the user receives the a URL from the same domain again, then for that particular URL, physical web notification can be treated as a priority notification.

**Comment by raggi** [comment]
I want priority notifications. Users will never seek this out, IMO.

**Comment by adriancretu** [comment]
I kind of have to agree that if no indicator of "something" is available on the notification bar, nobody will bother to pull down to see if there is anything there. I think what was previously classified as a "bug" was more like a feature. The main issue is - does Android allow showing a small icon if the notification is min-priority? In my actual experience with a test app, creating a "noticeable" min-priority notification is a miss-and-hit: under debug mode, an icon will show up, but under release mode, it will not.

**Comment by Hoopstr42** [comment]
It seems to me that the best solution would be to classify it as a default priority notification but have the default settings set to no sound/vibe and let the user decide if they want to be notified. @adriancretu is right, no one will ever pull down the bar unless there is a notification in the top bar.

**Comment by scottjenson** [comment]
This is a constant issue and a tension we see often: devs want to push a
hard notification and in our tests, users often feel it is a bit spammy. We
agree that having a MIN priority notification is a bit weak as most users
won't naturally find it. We expect, over time, as this is used
repetitively, users will get it but we totally understand this will take
time.

However, a high priority 'buzzing' notification isn't the answer. It is, of
course, helpful in the short term, but it really irritates users so we're
just never going to do that.

We're trying to find the right mix of physical prompts (e.g. using the
Physical Web logo in store) to help users know that there is something
nearby. This is how wifi usually works in coffee shops, users see the wifi
logo and know that can open their laptops.

**Comment by raggi** [comment]
My use case is in my house. I have aggregated public access for certain things (music, lights, garage, gate). I don't really want to plaster my house with stickers or explain to people anything. The promise of the physical web (to me) is exactly that zero comms, zero config, Zeo install. Anything that detracts from those devalues it. I'd almost be as well off putting a QR code or something on the wall, if it's not obviously discovered. I certainly sympathize to the spam concern, and I don't know if I have a good solution, but I think that trying to eradicate this factor is critical to success. QR codes did not work as well as people hoped because they required users to know and do things proactively. Maybe there are some informal / per implementor substandard a that could be applied, with a more open public standard. The example that comes to mind is AMP, which enforces some control beyond its standard by proxy to reduce abuse/enforce rules.

**Comment by scottjenson** [comment]
Well, for your house, I think that would fall in my 'repetitive use case'
so it's really clear what you'd need to do after just a few times using it.
And, of course, I'm not suggesting that you should have Physical Web logos
all over your house ;-) So, if I understand you properly, you'd like a way
for visitors to be prompted? My concern though is that *you'd* be prompted
as well, which I expect would start to get very tedious after even a few
days.

Scott

**Comment by raggi** [comment]
I'd only expect to be notified when I get home. That already often happens anyway, for things like location reminders. I'm totally ok with a buzz in my pocket as I get home.

**Comment by scottjenson** [comment]
I appreciate that you don't mind that but we've got a lot of user research
that shows quite the opposite unfortunately. Repetitive notifications are a
big issue to a large number of users.

Think of it this way, you're phone doesn't buzz evertime you see a few wifi
hotspot. We are much the same way: we'll show you something if you ask. I
appreciate this puts the effort on the users part, but it seems to be the
only way to do it without irritating them.

**Comment by matisay** [comment]
Hi @scottjenson 
I agree that repetitive notifications are big issue for the users That is right. For me one side is QRcode the other side SMS and the physicl web is in the middle. If we compare QRcode vs physical web, physical web is  very strong and easy to use. But SMS vs Physical Web has many weakness. But there is one big strongest part to the SMS and that is "affect the behavior right in the desicion". To use this, there is only weapon is "pushing a notification".
If a user download an app and "turn on the bluetooth (which is a very big issue for us in Turkey for common wrong opinion)" we can assume that he/she is willing to have notifications. beacuse he/she knows that if the bluetoth is on beaware of campaigns.
It is my oppinion. please consider.

**Comment by ReviTULize** [comment]
A Nearby or Physical Web notification(silent) will certainly speed adoption and consumer education.  I ask everyone I come in contact with if they know what this[physical web/nearby] symbol is and noone knows it yet. 

A silent notification would be ideal...to increase awareness.  Same as WiFi...It connects automatically(project fi), but then puts a questions mark stating that I need to 'do something'.
